src/storage/src/source/postgres.rs

```diff
@@ -849,14 +887,17 @@ impl PostgresTaskInfo {
                                 TupleData::UnchangedToast => old,
                                 _ => new,
                             });
+                        let new_datums =
+                            try_fatal!(PostgresTaskInfo::datums_from_tuple(rel_id, new_tuple));
                         let new_row =
-                            try_fatal!(PostgresTaskInfo::row_from_tuple(rel_id, new_tuple));
-                        inserts.push(new_row);
+                            try_fatal!(PostgresTaskInfo::cast_row(table_cast, &new_datums));
+                        inserts.push((output, new_row));
                     }
                     Delete(delete) if self.source_tables.contains_key(&delete.rel_id()) => {
                         last_data_message = Instant::now();
                         self.metrics.deletes.inc();
                         let rel_id = delete.rel_id();
+                        let &(output, _, ref table_cast) = self.source_tables.get(&rel_id).unwrap();
```

Comment:
do we tell pg in the replication stream what tables we want?

Reply:
No, pg will give all tables of a publication. We don't push down any stream demand info either. That's an optimization we could do in the future

Reply:
nice